### PR TITLE
refactor: recover task status (#124)

### PR DESCRIPTION
**Pull request by captainlee1024**

### CODE REVIEW

File: service/proto/src/proto/prover/v1/prover.proto [hunk 1]
```diff
@@ -102,8 +145,6 @@ message GenFinalProofResponse {
 message FinalProof {
   string proof = 1;
   string public_input = 2;
-  bytes pre_state_root = 3;
```
Comment (eigmax) [inline service/proto/src/proto/prover/v1/prover.proto:105]: We should reserve those two fields.

Comment (captainlee1024) [reply service/proto/src/proto/prover/v1/prover.proto:105]: They are now saved to the Zeth side.

File: service/src/prover_service.rs [hunk 2]
```diff
@@ -122,24 +128,59 @@ impl ProverService for ProverServiceSVC {
...
+                                match batch_req {
+                                    Step::GenBatchChunks(gen_chunk_req) => {
+                                        handler_clone
+                                            .handle_gen_batch_chunks_request(
+                                                request_id.clone(), gen_chunk_req, eth_client_clone.clone()
+                                            ).await.unwrap_or_else(|e| ProverResponse {
+                                            id: request_id.clone(),
+                                            response_type: Some(ResponseType::GenBatchProof(
+                                                GenBatchProofResponse {
+                                                    step: Some(gen_batch_proof_response::Step::GenBatchChunks(GenBatchChunksResult{
```
Comment (eigmax) [inline service/src/prover_service.rs:142]: GenBatchChunksResult{
result_code: ProofResultCode::CompletedError as i32,
rror_message: e.to_string(),
..Defult::default()
}

Do the same for other error handlers.

Comment (captainlee1024) [reply service/src/prover_service.rs:142]: ok

**Review by eigmax — APPROVED** [review]

**Pull request merged**